src/CoreServices/FSEvents.cs

```diff
@@ -307,16 +344,35 @@ public void ScheduleWithRunLoop (NSRunLoop runLoop)
 			ScheduleWithRunLoop (runLoop.GetCFRunLoop (), CFRunLoop.ModeDefault);
 		}
 
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern void FSEventStreamUnscheduleFromRunLoop (IntPtr handle,
+			IntPtr runLoop, IntPtr runLoopMode);
+
+		public void UnscheduleFromRunLoop (CFRunLoop runLoop, NSString runLoopMode)
+		{
+			CheckDisposed ();
+			if (runLoopMode == null) {
+				throw new ArgumentNullException (nameof (runLoopMode));
+			}
+			FSEventStreamUnscheduleFromRunLoop (handle, runLoop.GetHandle (), runLoopMode.GetHandle ());
+		}
+
+		public void UnscheduleFromRunLoop (CFRunLoop runLoop) => UnscheduleFromRunLoop (runLoop, CFRunLoop.ModeDefault);
+
+		public void UnscheduleFromRunLoop (NSRunLoop runLoop, NSString runLoopMode) => UnscheduleFromRunLoop (runLoop.GetCFRunLoop (), runLoopMode);
+
+		public void UnscheduleFromRunLoop (NSRunLoop runLoop) => UnscheduleFromRunLoop (runLoop.GetCFRunLoop (), CFRunLoop.ModeDefault);
+
 		[DllImport (Constants.CoreServicesLibrary)]
 		static extern IntPtr FSEventStreamCopyPathsBeingWatched (IntPtr handle);
 
 		public string [] PathsBeingWatched {
 			get {
 				CheckDisposed ();
-				var cfarray = new CFArray (FSEventStreamCopyPathsBeingWatched (handle), true);
+				using var cfarray = new CFArray (FSEventStreamCopyPathsBeingWatched (handle));
```

Comment:
This looks wrong, according to the `Copy` rule you own the return value from `FSEventStreamCopyPathsBeingWatched `, I think it should be like this:

```suggestion
				using var cfarray = new CFArray (FSEventStreamCopyPathsBeingWatched (handle), true);
```